### PR TITLE
CI: Attempt to fix wheel builds

### PR DESCRIPTION
**Pull request by lithomas1**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by mroeschke** [comment]
Since the repo is cloned into this job with the sdist, for debugging it may be worth checking that the key files (project.toml, setup.py, meson files) are synced between both

**Comment by lithomas1** [comment]
> Since the repo is cloned into this job with the sdist, for debugging it may be worth checking that the key files (project.toml, setup.py, meson files) are synced between both

Can we put this in as a band-aid for now, and see if it fixes the nightlies?

I don't think I have the energy/time to repro/root cause this.
(more details in the comment I pinged you on before.
TLDR: only fails on nightlies. succeeds when you re-run failed builds. has never failed on a manually run workflow or a PR workflow).

**Review by mroeschke — APPROVED** [review]
